CHANGELOG.md

```diff
@@ -46,6 +46,8 @@ Main (unreleased)
 
 - Fix a race condition where the ui service was dependent on starting after the remotecfg service, which is not guaranteed. (@dehaansa & @erikbaranowski)
 
+- `loki.source.podlogs`: Fixed a bug which prevented clustering from working and caused duplicate logs to be sent. (@ptodev)
```

Comment:
This was not working only in the case where the clustering config would be updated at runtime and without changing the selectors, right? On restart of Alloy it would work, no?

Reply:
It depends. If you're using default configs like this, then clustering will never work:
```
loki.source.podlogs "podlogs_objects" {
  client {
    kubeconfig_file = "/Users/paulintodev/.kube/config"
  }
  clustering {
    enabled = true
  }
  forward_to = [loki.echo.example.receiver]
}
```

That's because on startup, the "new" and the "old" configs are both the same.

Reply:
I updated the readme to mention this more clearly :) I haven't tested what happens when a `selector` or `namespace_selector` are specified, but I'm pretty sure the code will work ok from what I can see